test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -332,3 +488,152 @@ func checkData(t *testing.T, start time.Time, lineCount int) {
 	)
 	assert.NoError(t, err)
 }
+
+func ValidateEntity(t *testing.T, logGroup, logStream string, end *time.Time, expectedEntity expectedEntity) {
+	log.Printf("Validating entity for log group: %s, stream: %s", logGroup, logStream)
+
+	logGroupInfo, err := getLogGroup()
+	for _, lg := range logGroupInfo {
+		if *lg.LogGroupName == logGroup {
+			log.Println("Log group " + *lg.LogGroupName + " exists")
+			break
+		}
+	}
+	assert.NoError(t, err)
+	begin := end.Add(-sleepForExtendedFlush * 4)
```

Comment:
Instead of trying to calculate the begin time, can we just pass in the begin time?

Reply:
Updated this for `ValidateEntity()` (commit [7950725](https://github.com/aws/amazon-cloudwatch-agent-test/pull/433/commits/795072522768ef075f3068f93a5c84642193acc3))